ios/l2_interfaces/ios_l2_interface.yaml

```diff
@@ -0,0 +1,74 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: ios
+RESOURCE: l2_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: ios_l2_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-2 interface on Cisco IOS devices.
+  description: This module provides declarative management of Layer-2 interface on Cisco IOS devices.
+  author: Sumit Jaiswal (@justjais)
+  options:
+    config:
+      description: A dictionary of Layer-2 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of interface, e.g. ge-0/0/0.
+          type: str
+          required: True
+        mode:
```

Comment:
What do you think about just having "access" and "trunk" keys with their config nested.  If we approach it like this, we can eliminate the "mode" and require one of "access" or "trunk"